java-checks/src/test/files/checks/SwitchDefaultLastCaseCheck.java

```diff
@@ -1,29 +1,59 @@
 class A {
-  
+
   void foo() {
     switch (0) {
-    default:   // Noncompliant [[sc=5;ec=13]] {{Move this default to the end of the switch.}}
-    case 0:
-      break;
+      case 0:
+      default: // Noncompliant [[sc=7;ec=15]] {{Move this default to the end of the switch.}}
+        break;
+      case 1:
+        break;
     }
-  
-    switch(0) {
-      case 0:     // Compliant
+
+    switch (0) {
+      case 0: // Compliant
         break;
     }
-    
-    switch(0) {
-      default:  // Compliant
+
+    switch (0) {
+      default: // Compliant
         break;
     }
-    
-    switch(0) {
+
+    switch (0) {
+      case 0:
+        break;
+      default: // Noncompliant
+        break;
+      case 1:
+        break;
+    }
+
+    switch (0) {
+      default: // Compliant
+      case 0:
+        break;
+      case 1:
+        break;
+    }
+
+    switch (0) {
+      default: // Noncompliant
+        break;
       case 0:
         break;
-      default:  // Noncompliant
+      case 1:
+        break;
+    }
+    
+    switch (0) {
+      case 0:
         break;
       case 1:
         break;
+      case 2:
+      default:  // Compliant
```

Comment:
This one is interesting, and it deserve a special comment -> why it's compliant (no need 2 lines of text, just a few word, such as "will not have any effect, as requires to match '3' ")